### ISSUE TITLE
在用RAVQA做测试时，每个passages的输出时下边这样的。这正常吗？

### ISSUE THREAD

**Issue by 1286169349** [LinWeizheDragon/Retrieval-Augmented-Visual-Question-Answering#20]
[PAD] race
[PAD] race
[PAD] race  [unused12] [unused12]
总是存在一些奇怪的token，请问怎么回事呢？

**Comment by LinWeizheDragon** [comment]
如果正常训练应该不会出现这个情况   你用的是T5吗？
出现这种tokens可能说明没有训练好，如果你修改了Loss function的话  需要仔细检查一下loss function是否有训练模型输出closing token （比如<eos>）

**Comment by 1286169349** [comment]
用的是T5-large.而且测试时的输出是下边这样的。应该也不对吧
Testing DataLoader 0:   0%|          | 6/2523 [00:35<4:07:42,  5.90s/it]artist <--->     (<PAD> [SEP] <PAD> [SEP] <PAD> [SEP])
condiment <--->     (<PAD> [SEP] <PAD> [SEP] <PAD>)
Testing DataLoader 0:   0%|          | 7/2523 [00:41<4:05:43,  5.86s/it]work <--->     (<PAD> <PAD> <PAD> <PAD> <PAD> <PAD>)
bmx <--->     (<PAD> [SEP] <PAD> [SEP] <PAD> [SEP])
Testing DataLoader 0:   0%|          | 8/2523 [00:46<4:06:01,  5.87s/it]commercial <--->     (<PAD> <PAD> <PAD> <PAD> <PAD> <PAD>)
girl <---> [unused12] ∧ [unused12]    (<PAD> [unused12] ∧ [SEP] <PAD> [unused12])
Testing DataLoader 0:   0%|          | 9/2523 [00:52<4:06:06,  5.87s/it]1847 <--->     (<PAD> [SEP] <PAD> [SEP] <PAD>)
ride <--->     (<PAD> [SEP] <PAD> [SEP] <PAD> [SEP])
Testing DataLoader 0:   1%|          | 29/2523 [02:46<3:59:08,  5.75s/it]
进程已结束,退出代码-1

**Comment by 1286169349** [comment]
我没有修改代码，只是用了从huggingface上下载的T5-large模型

**Comment by 1286169349** [comment]
好像是因为移位操作导致每个输出前都有一个[PAD],请问这是正确的吗，还是代码应该改为左移后去掉[PAD]的
shifted_generation_outputs[:, :-1] = generation_outputs[:, 1:]#左移去掉[PAD]，这一步中的shifted_generation_outputs，后边并没有使用。
是不是应该将下边的(rag_model.py)中的generation_outputs改为shifted_generation_outputs
                for b in range(batch_size):
                    # use topk to get indices of top candidates
                    top_cand_inds = (-loss_with_doc_scores[b]).topk(1)[1]
                    outputs.append(generation_outputs[b, top_cand_inds])
                    answer_proposals = generation_outputs_decoded[b*n_docs:(b+1)*n_docs]
                    generation_outputs_for_docs.append(answer_proposals)
                    # print(-loss[b])
                    # print(answer_proposals)
                outputs = torch.cat(outputs)

**Comment by 1286169349** [comment]
> 如果正常训练应该不会出现这个情况 你用的是T5吗？ 出现这种tokens可能说明没有训练好，如果你修改了Loss function的话 需要仔细检查一下loss function是否有训练模型输出closing token （比如）

可以帮助我吗？万分感谢
